pkgs/development/python-modules/wxpython/4.2.nix

```diff
@@ -40,11 +41,16 @@
   six,
 }:
 
+let
+  waf_2_0_25 = fetchurl {
```

Comment:
Is this this waf https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/wa/waf/package.nix ? Can we just use that instead?

Reply:
Yes, but 2.1.2 is too recent, wxPython doesn't build with it.

Reply:
@wegank are you sure that wxPython doesn't build with waf 2.1.2? I just tried (see [this branch](https://github.com/jfly/nixpkgs/tree/wxpython-fix)), and `nix build .#python3.pkgs.wxpython` seems to work for me.

Is it possible you meant to say that wxPython doesn't build with 2.1.1? At the time of writing, this PR (#326142)'s commit is 9dc9813a8d49817a36bd0d11ff57417c469102aa, whose parent is 7c1d59ff54058cf7b36ef0aea0aad7a0870a17f7, which [has waf 2.1.1](https://github.com/jfly/nixpkgs/blob/7c1d59ff54058cf7b36ef0aea0aad7a0870a17f7/pkgs/by-name/wa/waf/package.nix#L13), so I'm a little surprised to see you mention waf 2.1.2 here. I tried building wxPython with waf 2.1.1, and it pretty quickly errors out with a conflict in argparse about a `--python` option, which was reported upstream [here](https://gitlab.com/ita1024/waf/-/merge_requests/2371), and fixed in [this commit](https://gitlab.com/ita1024/waf/-/commit/bba5047716ea26d93f501a30a4c6aa0ce8c1642a), which first landed in waf 2.1.2.

tl;dr: nixpkgs now has [waf 2.1.2](https://github.com/NixOS/nixpkgs/pull/328237). If you rebase this PR, I am pretty sure you'll find that you can follow @SuperSandro2000's advice and just use `waf`.

Tangentially related, I found a (IMO) cleaner mechanism for overriding waf, which you can see in my branch: https://github.com/NixOS/nixpkgs/compare/master...jfly:nixpkgs:wxpython-fix. Feel free to steal it. Also happy to make a PR out of my branch instead you prefer.